fitbenchmarking/results_processing/plots.py

```diff
@@ -68,182 +45,122 @@ def __init__(self, best_result, options, figures_dir):
-        plot_options_dict['color'] = self.best_fit_plot_options['color']
-
-        y = result.fin_y[result.sorted_index]
-        self.plot_data(errors=False,
-                       plot_options=plot_options_dict,
-                       y=y)
-        self.format_plot()
-        file = f"{result.sanitised_min_name(True)}_fit_for_" \
-               f"{self.result.costfun_tag}_{self.result.sanitised_name}.png"
-        file_name = os.path.join(self.figures_dir, file)
-        self.fig.savefig(file_name)
-
-        # Update to correct linestyle
-        plot_options_dict = self.best_fit_plot_options.copy()
-        plot_options_dict['label'] = label
-        self.plot_data(errors=False,
-                       plot_options=plot_options_dict,
-                       x=self.x,
-                       y=y)
-
-        # Make sure line wont be replaced by resetting line_plot
-        self.line_plot = None
-        return file
-
-    def plot_fit(self, result):
+        htmlfile = f"{result.sanitised_min_name(True)}_fit_for_"\
+            f"{self.result.costfun_tag}_{self.result.sanitised_name}.html"
+        return htmlfile
+
+    def plotly_fit(self, df):
         """
-        Updates self.line to show the fit using the passed in params.
-        If self.line is empty it will create a new line.
+        Uses plotly to plot the calculated fit, along with the best fit.
         Stores the plot in a file
 
-        :param result: The result to plot
-        :type result: FittingResult
+        :param df: A dataframe holding the data
+        :type df: Pandas dataframe
 
         :return: path to the saved file
         :rtype: str
         """
-        plot_options_dict = self.fit_plot_options.copy()
-        plot_options_dict['label'] = result.modified_minimizer_name(True)
-
-        self.plot_data(errors=False,
-                       plot_options=plot_options_dict,
-                       x=self.x,
-                       y=result.fin_y[result.sorted_index])
-        self.format_plot()
-        file = f"{result.sanitised_min_name(True)}_fit_for_"\
-               f"{self.result.costfun_tag}_{self.result.sanitised_name}.png"
-        file_name = os.path.join(self.figures_dir, file)
-        self.fig.savefig(file_name)
-        return file
+        # Plotly implementation below
+        htmlfiles = {}
+        for minimizer in df['minimizer'].unique():
+            if minimizer not in ["Data", "Starting Guess"]:
+                fig = px.line(df.query(f"minimizer == '{minimizer}'"),
+                              x="x",
+                              y="y",
+                              color="minimizer",
+                              title=self.result.name,
+                              markers=True)
+                if not df.query(f"minimizer == '{minimizer}'")["best"].any():
+                    # add the best plot
+                    name = 'Best Fit (' + \
+                        f'{df.query("best")["minimizer"].unique()})'
```

Comment:
This seems to give an odd name -- I think unique is returning a list:
![image](https://github.com/fitbenchmarking/fitbenchmarking/assets/48282025/29be9354-d53f-497d-a43c-c3710105c810)

By construction will this only ever have 1 value? If so can we just take the first one? And in that case do we need the above `any()` as that's also just the same as the first value?